src/otx/algo/utils/xai_utils.py

```diff
@@ -2,58 +2,83 @@
 # SPDX-License-Identifier: Apache-2.0
 """Utils used for XAI."""
 
+# TODO(someone): Typings in this file is too weak or wrong. It should be fixed.
+# For example, `pred_labels: list | None` has no object typing containered in the list.
+# On the other hand, process_saliency_maps should not produce list of dictionaries
+# (`list[dict[str, np.ndarray | torch.Tensor]]`).
+# This is because the output will be assigned to OTXBatchPredEntity.saliency_map,
+# but this has `list[np.ndarray | torch.Tensor]` typing, so that it makes a mismatch.
```

Comment:
Dict entity is needed not to lose the saliency map class index after selecting a subset of saliency map with selected class 
`{class_idx: class_saliency_map}`

I'll need to look into that

Reply:
The problem is that the output goes into `OTXBatchPredEntity.saliency_map` which has `list[np.ndarray | torch.Tensor]` typing. My concern is that mistyping in runtime should not exist.

Reply:
Yes, this makes sense